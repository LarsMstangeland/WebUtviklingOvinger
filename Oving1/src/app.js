export function isLeapYear(year){

    //oppgave 2.1
    if(!(year > 0)){
        throw new InvalidArgumentError ('Invalid argument: year must be an integer equal or larger than 0')
    }

    //oppgave 3.1
    if((year == null) || (year == undefined)){
        throw new InvalidArgumentError ('Invalid argument: year cannot be null or undefinded')
    }

    //Oppgave 1
    return  (year % 4 == 0) && 
            (year % 100 !== 0) || 
            (year % 400 == 0);
}

