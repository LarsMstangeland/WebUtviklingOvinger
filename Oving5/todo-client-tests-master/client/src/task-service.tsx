import axios from 'axios';

axios.defaults.baseURL = 'http://localhost:3000/api/v2';

export type Task = {
  id: number;
  title: string;
  done: boolean;
  description: string;
};

class TaskService {
  /**
   * Get task with given id.
   */
  get(id: number) {
    return axios.get<Task>('/tasks/' + id).then((response) => response.data);
  }

  /**
   * Get all tasks.
   */
  getAll() {
    return axios.get<Task[]>('/tasks').then((response) => response.data);
  }

  /**
   * Create new task having the given title.
   *
   * Resolves the newly created task id.
   */
  create(title: string, description: string) {
    return axios
      .post<{ id: number }>('/tasks', { title: title, description: description })
      .then((response) => response.data.id);
  }

  delete(id: number){
    return axios.delete('/tasks/'+id).then((response) => response.data)
  }

  update(task:Task){
    return axios.patch('/tasks/'+task.id, task).then((response) => response.data)
  }

}



const taskService = new TaskService();
export default taskService;
