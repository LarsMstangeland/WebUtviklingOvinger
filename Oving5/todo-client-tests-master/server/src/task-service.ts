import pool from './mysql-pool';
import type { RowDataPacket, ResultSetHeader } from 'mysql2';

export type Task = {
  id: number;
  title: string;
  done: boolean;
  description: string;
};

class TaskService {
  /**
   * Get task with given id.
   */
  get(id: number) {
    return new Promise<Task | undefined>((resolve, reject) => {
      pool.query('SELECT * FROM Tasks WHERE id = ?', [id], (error, results: RowDataPacket[]) => {
        if (error) return reject(error);

        resolve(results[0] as Task);
      });
    });
  }

  /**
   * Get all tasks.
   */
  getAll() {
    return new Promise<Task[]>((resolve, reject) => {
      pool.query('SELECT * FROM Tasks', (error, results: RowDataPacket[]) => {
        if (error) return reject(error);

        resolve(results as Task[]);
      });
    });
  }

  /**
   * Create new task having the given title.
   *
   * Resolves the newly created task id.
   */
  create(title: string, description: string) {
    return new Promise<number>((resolve, reject) => {
      pool.query('INSERT INTO Tasks SET title=?, description=?', [title,description], (error, results: ResultSetHeader) => {
        if (error) return reject(error);
        resolve(results.insertId);
      });
    });
  }

  /**
   * Delete task with given id.
   */
  delete(id: number) {
    return new Promise<void>((resolve, reject) => {
      pool.query('DELETE FROM Tasks WHERE id = ?', [id], (error, results: ResultSetHeader) => {
        if (error) return reject(error);
        if (results.affectedRows == 0) return reject(new Error('No row deleted'));
        resolve();
      });
    });
  }

  update(task:Task){
    return new Promise<void>((resolve, reject) => {
      pool.query('UPDATE Tasks SET done=?, description=? WHERE id=?', [task.done, task.description, task.id], (error, results: ResultSetHeader) => {
        if(error) return reject(error)
        resolve();
      });
    });
  }
}

const taskService = new TaskService();
export default taskService;
