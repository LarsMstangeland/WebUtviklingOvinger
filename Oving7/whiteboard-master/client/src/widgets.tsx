import * as React from 'react';
import { ReactNode, ChangeEvent } from 'react';
import { Component } from 'react-simplified';
import { NavLink } from 'react-router-dom';

/**
 * Renders an information card using Bootstrap classes.
 *
 * Properties: title
 */
export class Card extends Component<{ title: ReactNode }> {
  render() {
    return (
      <div className="card">
        <div className="card-body">
          <h5 className="card-title">{this.props.title}</h5>
          <div className="card-text">{this.props.children}</div>
        </div>
      </div>
    );
  }
}

/**
 * Renders a row using Bootstrap classes.
 */
export class Row extends Component {
  render() {
    return <div className="row">{this.props.children}</div>;
  }
}

/**
 * Renders a column with specified width using Bootstrap classes.
 *
 * Properties: width, right
 */
export class Column extends Component<{ width?: number; right?: boolean }> {
  render() {
    return (
      <div className={'col' + (this.props.width ? '-' + this.props.width : '')}>
        <div className={'float-' + (this.props.right ? 'end' : 'start')}>{this.props.children}</div>
      </div>
    );
  }
}

/**
 * Renders a success button using Bootstrap styles.
 *
 * Properties: small, onClick
 */
class ButtonSuccess extends Component<{ small?: boolean; onClick: () => void }> {
  render() {
    return (
      <button
        type="button"
        className="btn btn-success"
        style={
          this.props.small
            ? {
                padding: '5px 5px',
                fontSize: '16px',
                lineHeight: '0.7',
              }
            : {}
        }
        onClick={this.props.onClick}
      >
        {this.props.children}
      </button>
    );
  }
}

/**
 * Renders a danger button using Bootstrap styles.
 *
 * Properties: small, onClick
 */
class ButtonDanger extends Component<{ small?: boolean; onClick: () => void }> {
  render() {
    return (
      <button
        type="button"
        className="btn btn-danger"
        style={
          this.props.small
            ? {
                padding: '5px 5px',
                fontSize: '16px',
                lineHeight: '0.7',
              }
            : {}
        }
        onClick={this.props.onClick}
      >
        {this.props.children}
      </button>
    );
  }
}

/**
 * Renders a light button using Bootstrap styles.
 *
 * Properties: small, onClick
 */
class ButtonLight extends Component<{ small?: boolean; onClick: () => void }> {
  render() {
    return (
      <button
        type="button"
        className="btn btn-light"
        style={
          this.props.small
            ? {
                padding: '5px 5px',
                fontSize: '16px',
                lineHeight: '0.7',
              }
            : {}
        }
        onClick={this.props.onClick}
      >
        {this.props.children}
      </button>
    );
  }
}

/**
 * Renders a button using Bootstrap styles.
 *
 * Properties: onClick
 */
export class Button {
  static Success = ButtonSuccess;
  static Danger = ButtonDanger;
  static Light = ButtonLight;
}

/**
 * Renders a NavBar link using Bootstrap styles.
 *
 * Properties: to
 */
class NavBarLink extends Component<{ to: string }> {
  render() {
    return (
      <NavLink className="nav-link" activeClassName="active" to={this.props.to}>
        {this.props.children}
      </NavLink>
    );
  }
}

/**
 * Renders a NavBar using Bootstrap classes.
 *
 * Properties: brand
 */
export class NavBar extends Component<{ brand: ReactNode }> {
  static Link = NavBarLink;

  render() {
    return (
      <nav className="navbar navbar-expand-sm navbar-light bg-light">
        <div className="container-fluid justify-content-start">
          <NavLink className="navbar-brand" activeClassName="active" exact to="/">
            {this.props.brand}
          </NavLink>
          <div className="navbar-nav">{this.props.children}</div>
        </div>
      </nav>
    );
  }
}

/**
 * Renders a form label using Bootstrap styles.
 */
class FormLabel extends Component {
  render() {
    return <label className="col-form-label">{this.props.children}</label>;
  }
}

/**
 * Renders a form input using Bootstrap styles.
 */
class FormInput extends Component<{
  type: string;
  value: string | number;
  onChange: (event: ChangeEvent<HTMLInputElement>) => void;
  [prop: string]: any;
}> {
  render() {
    // ...rest will contain extra passed attributes such as disabled, required, width, height, pattern
    // For further information, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax
    const { type, value, onChange, ...rest } = this.props;
    return (
      <input
        {...rest}
        className="form-control"
        type={this.props.type}
        value={this.props.value}
        onChange={this.props.onChange}
      />
    );
  }
}

/**
 * Renders a form textarea using Bootstrap styles.
 */
class FormTextarea extends React.Component<{
  value: string | number;
  onChange: (event: ChangeEvent<HTMLTextAreaElement>) => void;
  [prop: string]: any;
}> {
  render() {
    // ...rest will contain extra passed attributes such as disabled, required, rows, cols
    // For further information, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax
    const { value, onChange, ...rest } = this.props;
    return <textarea {...rest} className="form-control" value={value} onChange={onChange} />;
  }
}

/**
 * Renders a form checkbox using Bootstrap styles.
 */
class FormCheckbox extends Component<{
  checked: boolean;
  onChange: (event: ChangeEvent<HTMLInputElement>) => void;
  [prop: string]: any;
}> {
  render() {
    // ...rest will contain extra passed attributes such as disabled, required, width, height, pattern
    // For further information, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax
    const { checked, onChange, ...rest } = this.props;
    return (
      <input
        {...rest}
        className="form-check-input"
        type="checkbox"
        checked={checked}
        onChange={onChange}
      />
    );
  }
}

/**
 * Renders a form select using Bootstrap styles.
 */
class FormSelect extends Component<{
  value: string | number;
  onChange: (event: ChangeEvent<HTMLSelectElement>) => void;
  [prop: string]: any;
}> {
  render() {
    // ...rest will contain extra passed attributes such as disabled, required, size.
    // For further information, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax
    const { value, onChange, children, ...rest } = this.props;
    return (
      <select {...rest} className="custom-select" value={value} onChange={onChange}>
        {children}
      </select>
    );
  }
}

/**
 * Renders form components using Bootstrap styles.
 */
export class Form {
  static Label = FormLabel;
  static Input = FormInput;
  static Textarea = FormTextarea;
  static Checkbox = FormCheckbox;
  static Select = FormSelect;
}

/**
 * Renders alert messages using Bootstrap classes.
 *
 * Students: this slightly more complex component is not part of curriculum.
 */
export class Alert extends Component {
  alerts: { id: number; text: ReactNode; type: string }[] = [];
  nextId: number = 0;

  render() {
    return (
      <div>
        {this.alerts.map((alert, i) => (
          <div
            key={alert.id}
            className={'alert alert-dismissible alert-' + alert.type}
            role="alert"
          >
            {alert.text}
            <button
              type="button"
              className="btn-close btn-sm"
              onClick={() => this.alerts.splice(i, 1)}
            />
          </div>
        ))}
      </div>
    );
  }

  /**
   * Show success alert.
   */
  static success(text: ReactNode) {
    // To avoid 'Cannot update during an existing state transition' errors, run after current event through setTimeout
    setTimeout(() => {
      let instance = Alert.instance(); // Get rendered Alert component instance
      if (instance) instance.alerts.push({ id: instance.nextId++, text: text, type: 'success' });
    });
  }

  /**
   * Show info alert.
   */
  static info(text: ReactNode) {
    // To avoid 'Cannot update during an existing state transition' errors, run after current event through setTimeout
    setTimeout(() => {
      let instance = Alert.instance(); // Get rendered Alert component instance
      if (instance) instance.alerts.push({ id: instance.nextId++, text: text, type: 'info' });
    });
  }

  /**
   * Show warning alert.
   */
  static warning(text: ReactNode) {
    // To avoid 'Cannot update during an existing state transition' errors, run after current event through setTimeout
    setTimeout(() => {
      let instance = Alert.instance(); // Get rendered Alert component instance
      if (instance) instance.alerts.push({ id: instance.nextId++, text: text, type: 'warning' });
    });
  }

  /**
   * Show danger alert.
   */
  static danger(text: ReactNode) {
    // To avoid 'Cannot update during an existing state transition' errors, run after current event through setTimeout
    setTimeout(() => {
      let instance = Alert.instance(); // Get rendered Alert component instance
      if (instance) instance.alerts.push({ id: instance.nextId++, text: text, type: 'danger' });
    });
  }
}
