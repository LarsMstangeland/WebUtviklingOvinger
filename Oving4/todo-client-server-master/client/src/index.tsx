import ReactDOM from 'react-dom';
import * as React from 'react';
import { Component } from 'react-simplified';
import { Card, Row, Column, Form, Button } from './widgets';
import taskService, { Task } from './task-service';

class TaskList extends Component {
  tasks: Task[] = [];
  checkboxValue: boolean = false;

  render() {
    return (
      <Card title="Tasks">
        {this.tasks.map((task) => (

          <Row key={task.id}>
            <Column>{task.title}</Column>

            <Column>

              <Form.Checkbox checked={task.done} onChange={() => {
                taskService.updateDone(task.id, !task.done).then(() => {
                  this.mounted();
                })

              }}></Form.Checkbox>
            </Column>


            <Column>
              <Button.Danger small={true}
                onClick={() => {
                  taskService.delete(task.id).then(()=>
                  this.mounted());
                }}>
                  x
              </Button.Danger>
            </Column>
          </Row>
        ))}
      </Card>
    );
  }

  mounted() {
    taskService.getAll().then((tasks) => (this.tasks = tasks));
  }
}

class TaskNew extends Component {
  title = '';

  render() {
    return (
      <Card title="New task">
        <Row>
          <Column width={1}>
            <Form.Label>Title:</Form.Label>
          </Column>
          <Column width={4}>
            <Form.Input
              type="text"
              value={this.title}
              onChange={(event) => (this.title = event.currentTarget.value)}
            />
          </Column>
        </Row>
        <Button.Success
          onClick={() => {
            taskService.create(this.title).then(() => {
              // Reloads the tasks in the Tasks component
              TaskList.instance()?.mounted(); // .? meaning: call TaskList.instance().mounted() if TaskList.instance() does not return null
              this.title = '';
            });
          }}
        >
          Create
        </Button.Success>
      </Card>
    );
  }
}

ReactDOM.render(
  <>
    <TaskList />
    <TaskNew />
  </>,
  document.getElementById('root')
);
