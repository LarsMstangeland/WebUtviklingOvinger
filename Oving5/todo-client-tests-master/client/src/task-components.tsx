import * as React from 'react';
import { Component } from 'react-simplified';
import { Alert, Card, Row, Column, Form, Button } from './widgets';
import { NavLink } from 'react-router-dom';
import taskService, { Task } from './task-service';
import { createHashHistory } from 'history';

const history = createHashHistory(); // Use history.push(...) to programmatically change path, for instance after successfully saving a student

/**
 * Renders task list.
 */
export class TaskList extends Component {
  tasks: Task[] = [];

  render() {
    return (
      <>
        <Card title="Tasks">
          {this.tasks.map((task) => (
            <Row key={task.id}>
              <Column>
                <NavLink to={'/tasks/' + task.id}>{task.title}</NavLink>
              </Column>
            </Row>
          ))}
        </Card>
        <Button.Success onClick={() => history.push('/tasks/new')}>New task</Button.Success>
      </>
    );
  }

  mounted() {
    taskService
      .getAll()
      .then((tasks) => (this.tasks = tasks))
      .catch((error) => Alert.danger('Error getting tasks: ' + error.message));
  }
}

/**
 * Renders a specific task.
 */
export class TaskDetails extends Component<{ match: { params: { id: number } } }> {
  task: Task = { id: 0, title: '', done: false, description: '' };

  render() {
    return (
      <>
        <Card title="Task">
          <Row>
            <Column width={2}>Title:</Column>
            <Column>{this.task.title}</Column>
          </Row>
          <Row>
            <Column width={2}>Description:</Column>
            <Column>{this.task.description}</Column>
          </Row>
          <Row>
            <Column width={2}>Done:</Column>
            <Column>
              <Form.Checkbox checked={this.task.done} onChange={() => {}} disabled />
            </Column>
          </Row>
        </Card>
        <Button.Success
          onClick={() => history.push('/tasks/' + this.props.match.params.id + '/edit')}
        >
          Edit
        </Button.Success>
      </>
    );
  }

  mounted() {
    taskService
      .get(this.props.match.params.id)
      .then((task) => (this.task = task))
      .catch((error) => Alert.danger('Error getting task: ' + error.message));
  }
}

/**
 * Renders form to edit a specific task.
 */
export class TaskEdit extends Component<{ match: { params: { id: number } } }> {
  task: Task = { id: 0, title: '', done: false, description: '' };

  render() {
    return (
      <>
        <Card title="Edit task">
          <Row>
            <Column width={2}>
              <Form.Label>Title:</Form.Label>
            </Column>
            <Column>
              <Form.Input
                type="text"
                value={this.task.title}
                onChange={(event) => (this.task.title = event.currentTarget.value)}
              />
            </Column>
          </Row>
          <Row>
            <Column width={2}>
              <Form.Label>Description:</Form.Label>
            </Column>
            <Column>
              <Form.Textarea 
                type="text"
                value={this.task.description} 
                onChange={(event) => {
                  this.task.description = event.currentTarget.value
                }} 
                rows={10} 
                />
            </Column>
          </Row>
          <Row>
            <Column width={2}>Done:</Column>
            <Column>
              <Form.Checkbox
                checked={this.task.done}
                onChange={(event) => (this.task.done = event.currentTarget.checked)}
              />
            </Column>
          </Row>
        </Card>
        <Row>
          <Column>
            <Button.Success onClick={() => 
              taskService.update(this.task).then(() => history.push("/tasks"))
            }
            
            >Save</Button.Success>
          </Column>
          <Column right>
            <Button.Danger onClick={() => 
              taskService.delete(this.task.id).then(() => history.push('/tasks'))
            }
            >Delete</Button.Danger>
          </Column>
        </Row>
      </>
    );
  }

  mounted() {
    taskService
      .get(this.props.match.params.id)
      .then((task) => (this.task = task))
      .catch((error) => Alert.danger('Error getting task: ' + error.message));
  }
}

/**
 * Renders form to create new task.
 */
export class TaskNew extends Component {
  title = '';
  description = '';

  render() {
    return (
      <>
        <Card title="New task">
          <Row>
            <Column width={2}>
              <Form.Label>Title:</Form.Label>
            </Column>
            <Column>
              <Form.Input
                type="text"
                value={this.title}
                onChange={(event) => (this.title = event.currentTarget.value)}
              />
            </Column>
          </Row>
          <Row>
            <Column width={2}>
              <Form.Label>Description:</Form.Label>
            </Column>
            <Column>
              <Form.Textarea 
              value={this.description} 
              onChange={(event) => {
                this.description = event.currentTarget.value
              }} 
              rows={10} 
              />
            </Column>
          </Row>
        </Card>
        <Button.Success
          onClick={() => {
            taskService
              .create(this.title, this.description)
              .then((id) => history.push('/tasks/' + id))
              .catch((error) => Alert.danger('Error creating task: ' + error.message));
          }}
        >
          Create
        </Button.Success>
      </>
    );
  }
}
