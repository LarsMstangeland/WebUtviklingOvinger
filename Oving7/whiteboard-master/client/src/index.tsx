import ReactDOM from 'react-dom';
import * as React from 'react';
import { Whiteboard } from './whiteboard-component';
import { Alert } from './widgets';

ReactDOM.render(
  <>
    <Alert />
    <Whiteboard />
  </>,
  document.getElementById('root')
);
