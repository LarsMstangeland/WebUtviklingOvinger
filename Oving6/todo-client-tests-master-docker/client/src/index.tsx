import ReactDOM from 'react-dom';
import * as React from 'react';
import { Component } from 'react-simplified';
import { HashRouter, Route } from 'react-router-dom';
import { NavBar, Card, Alert, Form, Row, Button, Column } from './widgets';
import { TaskList, TaskDetails, TaskEdit, TaskNew } from './task-components';
import taskService from './task-service';

class Menu extends Component {
  render() {
    return (
      <NavBar brand="Todo App">
        <NavBar.Link to="/tasks">Tasks</NavBar.Link>
      </NavBar>
    );
  }
}

class Home extends Component {

  input: string = '';
  outputerror: string= '';
  outputgood: string = '';



  render() {
    return (
    <Card title="Welcome">This is a Code App
      <Row>
        <Form.Textarea rows={8} value={this.input} type={"text"} onChange={(event) => {

          this.input = event.currentTarget.value

        }}></Form.Textarea>
      </Row>
      <Row>
        <Button.Success
            onClick={() => {
              
              //console.info(this.input)

              taskService.run(this.input)
              .then((response) => {
                this.outputgood = response.data.stdout;
                this.outputerror = response.data.stderr;
              })
              /*
                .create(this.title)
                .then((id) => history.push('/tasks/' + id))
                .catch((error) => Alert.danger('Error Running code: '));
                */
            }}
          >
            Create
          </Button.Success>
      </Row>
      <Row>
          <Card title={"Result"}>
            <Column width={10}>No code here yet: {this.outputgood} </Column>
          </Card>
      </Row>
      <Row>
          <Card title={"Error"}>
            <Column width={10}>No code here yet: {this.outputerror}</Column>
          </Card>
      </Row>    
    </Card>);
  }
}

ReactDOM.render(
  <HashRouter>
    <div>
      <Alert />
      <Menu />
      <Route exact path="/" component={Home} />
      <Route exact path="/tasks" component={TaskList} />
      <Route exact path="/tasks/:id(\d+)" component={TaskDetails} /> {/* id must be number */}
      <Route exact path="/tasks/:id(\d+)/edit" component={TaskEdit} /> {/* id must be number */}
      <Route exact path="/tasks/new" component={TaskNew} />
    </div>
  </HashRouter>,
  document.getElementById('root')
);
