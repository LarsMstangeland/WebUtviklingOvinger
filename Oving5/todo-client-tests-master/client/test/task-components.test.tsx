import * as React from 'react';
import { TaskDetails, TaskEdit, TaskList, TaskNew } from '../src/task-components';
import { shallow } from 'enzyme';
import { Form, Button } from '../src/widgets';
import { NavLink } from 'react-router-dom';

jest.mock('../src/task-service', () => {
  class TaskService {
    getAll() {
      return Promise.resolve([
        { id: 1, title: 'Les leksjon', done: false },
        { id: 2, title: 'Møt opp på forelesning', done: false },
        { id: 3, title: 'Gjør øving', done: false },
      ]);
    }

    create() {
      return Promise.resolve(4); // Same as: return new Promise((resolve) => resolve(4));
    }

    get(){
      return Promise.resolve(
        {
          id:1, title: 'les leksjon', done: false
        }
      )
    }

    update() {
      return Promise.resolve(); // Same as: return new Promise((resolve) => resolve(4));
    }

    delete() {
      return Promise.resolve(4); // Same as: return new Promise((resolve) => resolve(4));
    }

  }
  return new TaskService();
});

describe('Task component tests', () => {
  test('TaskList draws correctly', (done) => {
    const wrapper = shallow(<TaskList />);

    // Wait for events to complete
    setTimeout(() => {
      expect(
        wrapper.containsAllMatchingElements([
          <NavLink to="/tasks/1">Les leksjon</NavLink>,
          <NavLink to="/tasks/2">Møt opp på forelesning</NavLink>,
          <NavLink to="/tasks/3">Gjør øving</NavLink>,
        ])
      ).toEqual(true);
      done();
    });
  });

  test('TaskList success button works', (done) => {

    const wrapper = shallow(<TaskList></TaskList>)

    wrapper.find(Button.Success).simulate('click')

    setTimeout(() => {
      expect(location.hash).toEqual('#/tasks/new')
      done();
    })

  })

  test('Tasknew draws correctly', (done) => {
    
    const wrapper = shallow(<TaskNew></TaskNew>)

    setTimeout(() => {

      expect(wrapper).toMatchSnapshot();
      done();
    });

  })

  test('TaskNew correctly sets location on create', (done) => {
    const wrapper = shallow(<TaskNew />);

    wrapper.find(Form.Input).simulate('change', { currentTarget: { value: 'Kaffepause' } });
    // @ts-ignore
    expect(wrapper.containsMatchingElement(<Form.Input value="Kaffepause" />)).toEqual(true);
    wrapper.find(Button.Success).simulate('click');
    // Wait for events to complete
    setTimeout(() => {
      expect(location.hash).toEqual('#/tasks/4');
      done();
    });
  });

  test('TaskDetails draws correctly', (done) => {

    const wrapper = shallow(<TaskDetails match={{params: {id:1}}}></TaskDetails>)
    setTimeout(() => {

      expect(wrapper).toMatchSnapshot();
      done();
    });
  });


  test('TaskEdit draws correctly', (done) => {

    const wrapper = shallow(<TaskEdit match={{params: {id:1}}}></TaskEdit>)
    
    wrapper.find(Form.Input).simulate('change', { currentTarget: { value: 'detailer' }});

    wrapper.find(Form.Textarea).simulate('change', { currentTarget: { value: 'merDetalier' }});

    wrapper.find(Form.Checkbox).simulate('change', {currentTarget: {checked: true}});


    
        
    // @ts-ignore
    expect(wrapper.containsMatchingElement(<Form.Input value='detailer' />)).toEqual(true)
    // @ts-ignore
    expect(wrapper.containsMatchingElement(<Form.Textarea value='merDetalier' />)).toEqual(true)
    // @ts-ignore
    expect(wrapper.containsMatchingElement(<Form.Checkbox checked={true} />)).toEqual(true)


    wrapper.find(Button.Success).simulate('click');
    wrapper.find(Button.Danger).simulate('click');


    setTimeout(() =>{

      expect(location.hash).toEqual('#/tasks');


      expect(wrapper).toMatchSnapshot();
      done();
    })

  })


  


});
