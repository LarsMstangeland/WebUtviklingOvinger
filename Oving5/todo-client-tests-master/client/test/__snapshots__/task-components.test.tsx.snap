// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task component tests TaskDetails draws correctly 1`] = `
<Fragment>
  <Card
    title="Task"
  >
    <Row>
      <Column
        width={2}
      >
        Title:
      </Column>
      <Column>
        les leksjon
      </Column>
    </Row>
    <Row>
      <Column
        width={2}
      >
        Description:
      </Column>
      <Column />
    </Row>
    <Row>
      <Column
        width={2}
      >
        Done:
      </Column>
      <Column>
        <FormCheckbox
          checked={false}
          disabled={true}
          onChange={[Function]}
        />
      </Column>
    </Row>
  </Card>
  <ButtonSuccess
    onClick={[Function]}
  >
    Edit
  </ButtonSuccess>
</Fragment>
`;

exports[`Task component tests TaskEdit draws correctly 1`] = `
<Fragment>
  <Card
    title="Edit task"
  >
    <Row>
      <Column
        width={2}
      >
        <FormLabel>
          Title:
        </FormLabel>
      </Column>
      <Column>
        <FormInput
          onChange={[Function]}
          type="text"
          value="les leksjon"
        />
      </Column>
    </Row>
    <Row>
      <Column
        width={2}
      >
        <FormLabel>
          Description:
        </FormLabel>
      </Column>
      <Column>
        <FormTextarea
          onChange={[Function]}
          rows={10}
          type="text"
        />
      </Column>
    </Row>
    <Row>
      <Column
        width={2}
      >
        Done:
      </Column>
      <Column>
        <FormCheckbox
          checked={false}
          onChange={[Function]}
        />
      </Column>
    </Row>
  </Card>
  <Row>
    <Column>
      <ButtonSuccess
        onClick={[Function]}
      >
        Save
      </ButtonSuccess>
    </Column>
    <Column
      right={true}
    >
      <ButtonDanger
        onClick={[Function]}
      >
        Delete
      </ButtonDanger>
    </Column>
  </Row>
</Fragment>
`;

exports[`Task component tests Tasknew draws correctly 1`] = `
<Fragment>
  <Card
    title="New task"
  >
    <Row>
      <Column
        width={2}
      >
        <FormLabel>
          Title:
        </FormLabel>
      </Column>
      <Column>
        <FormInput
          onChange={[Function]}
          type="text"
          value=""
        />
      </Column>
    </Row>
    <Row>
      <Column
        width={2}
      >
        <FormLabel>
          Description:
        </FormLabel>
      </Column>
      <Column>
        <FormTextarea
          onChange={[Function]}
          rows={10}
          value=""
        />
      </Column>
    </Row>
  </Card>
  <ButtonSuccess
    onClick={[Function]}
  >
    Create
  </ButtonSuccess>
</Fragment>
`;
